lib/bolt/pal.rb

```diff
@@ -521,10 +521,30 @@ def get_plan_mtime(path)
       end
     end
 
-    def convert_plan(plan_path)
+    def convert_plan(plan)
+      path = File.expand_path(plan)
+
+      # If the path doesn't exist, check if it's a plan name
+      unless File.exist?(path)
+        in_bolt_compiler do |compiler|
+          sig = compiler.plan_signature(plan)
+
+          # If the plan was loaded, look for it on the module loader
+          # There has to be an easier way to do this...
+          if sig
+            type = compiler.list_plans.find { |p| p.name == plan }
```

Comment:
Hmm I spent some time looking to see if there's a better way, but I think that might have diminishing returns - ultimately I don't think it's going to be a whole lot more efficient or readable than this :/

Reply:
I think you can do something like `Puppet.lookup(:loaders).private_environment_loader.find(Puppet::Pops::Parser::TypedName.new(:plan, plan))`